cmake/toolchain/clang.cmake

```diff
@@ -0,0 +1,6 @@
+set(CMAKE_SYSTEM_NAME Generic)
+set(CMAKE_C_COMPILER_WORKS 1)
+set(CMAKE_CXX_COMPILER_WORKS 1)
+
+set(CMAKE_C_LINK_EXECUTABLE "\"${CLANG++_BIN}\" ${LD_FLAGS} <CMAKE_C_LINK_FLAGS> <LINK_FLAGS> <OBJECTS> -o <TARGET> <LINK_LIBRARIES>")
```

Comment:
How will linking for EFI work here? The current [efi](https://github.com/Bareflank/hypervisor/blob/master/scripts/cmake/toolchain/clang_x86_64_efi.cmake) toolchain adds some pre and post commands for creating the EFI application

Reply:
For those linker flags in the EFI toolchain, I think we could convince CMake to add them in the right order using target properties.

However, this also brings up a larger issue of how to build/link anything that implements a loader for the vmm (windows drivers, linux drivers, arm bootloaders, etc). Most of those likely won't support building with clang/lld, so we might have to use ExternalProject to build those components. I'll try to add a follow up commit to this PR that handles building those as well.